src/util/file-finder.ts

```diff
@@ -55,6 +56,18 @@ export async function findMatchingClient(source: string, client: InstallableClie
         ghCore.debug(`${client} ${clientVersion} files: ${clientFiles.join(", ")}`);
     }
 
+    // In case of client being 'yq', executable and zip files both are present
+    // this will give warning to the users as multiple files will be found after
+    // filteration. So removing all the zipped files.
+    if (client === Inputs.YQ) {
+        clientFiles.forEach((file) => {
```

Comment:
this is a poor way of filtering.

To keep with the way the rest of the filters work, I'd add a filter to the filter pipelines below that is the opposite of `filterByZipped`, ie `filterByNotZipped`.

If I wanted to do it outside of the filter pipeline, I would do:

`clientFiles = clientFiles.filter((file) => !canExtract(file))`

Reply:
Yeah, this approach is far better. I will change that